virttest/qemu_devices/qcontainer.py

```diff
@@ -1061,16 +1061,23 @@ def pflash_handler(firmware_name, cmd):
                 first_image = images[0]
                 img_params = params.object_params(first_image)
                 img_params["backing_chain"] = "no"
-                img_info = json.loads(qemu_storage.QemuImg(img_params,
-                                                           current_data_dir,
-                                                           first_image).info(True, "json"))
-                img_path, img_name = os.path.split(img_info.get("filename"))
-
-                pflash_vars_name = "%s_%s_VARS.fd" % (self.vmname, img_name)
-                pflash_vars_path = os.path.join(img_path, pflash_vars_name)
-                # When image has backing files, treat it as a temporary image
-                if "backing-filename" in img_info:
-                    self.temporary_image_snapshots.add(pflash_vars_path)
+                img_obj = qemu_storage.QemuImg(img_params,
+                                               current_data_dir, first_image)
+                img_info = json.loads(img_obj.info(True, "json"))
+                if img_obj.is_remote_image():
+                    pflash_vars_name = "%s_%s_%s_%s_VARS.fd" % \
+                        (self.vmname, params['guest_name'],
+                         params['image_backend'], img_info.get("format"))
+                    pflash_vars_path = os.path.join(current_data_dir,
```

Comment:
Well,  whether `backend` is used as a field or creating a directory named `backend` is okay for me.
Just pay attention to `current_data_dir` may be `/var/lib/avocado/data/avocado-vt/` or `/mnt`, but we usually save images into `/var/lib/avocado/data/avocado-vt/images/`, so I am not sure if `os.path.join(current_data_dir, pflash_vars_name)` is the best way.
@luckyh @yanan-fu , what do you think?

Reply:
It is depends on if `images_base_dir` was defined,  I think `current_data_dir` is better than `data_dir.get_data_dir()`, thanks~

Reply:
@luckyh  Could you pls help review, thanks.